pytensor/graph/basic.py

```diff
@@ -936,6 +936,56 @@ def graph_inputs(
     yield from (r for r in ancestors(graphs, blockers) if r.owner is None)
 
 
+def explicit_graph_inputs(
+    graph: Variable[Any, Any] | Iterable[Variable[Any, Any]],
+) -> Generator[Variable, None, None]:
+    """
+    Get the root variables needed as inputs to a function that computes `graph`
+
+    Parameters
+    ----------
+    graph : TensorVariable
+        Output `Variable` instances for which to search backward through
+        owners.
+
+    Returns
+    -------
+    iterable
+        Generator of root Variables (without owner) needed to compile a function that evaluates `graphs`.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        import pytensor
+        import numpy as np
+        import pytensor.tensor as pt
+        from pytensor.graph.basic import explicit_graph_inputs
+
+        x = pt.vector('x')
+        y = pt.constant(2)
+        z = pt.mul(x*y)
+
+        out = list(explicit_graph_inputs(z))
+        f = pytensor.function([x], out)
```

Comment:
You got it backwards?
```suggestion
        inputs = list(explicit_graph_inputs(z))
        f = pytensor.function(inputs, z)
```